facebook/amend_monthly_microdata.R

```diff
@@ -47,12 +53,43 @@ amend_microdata <- function(input_dir, output_dir, static_dir, pattern = ".*[.]c
     # Drop any territories.
     data <- filter(data, !(state %in% c("AS", "GU", "PR", "VI", "MP")))
 
+    # what zip5 values have a large enough population (>100) to include in micro
+    # output. Those with too small of a population are blanked to NA
+    data <- blank_zips(data, valid_zips)
+
     # Save file under original name but in output directory.
     message("writing data for ", fname)
     write_csv(data, file.path(output_dir, fname))
   }
 }
 
+create_zip5 <- function(data) {
+  data$zip5 <- data$A3
+
+  # clean the ZIP data
+  data$zip5 <- stri_replace_all(data$zip5, "", regex = " *")
+  data$zip5 <- stri_replace(data$zip5, "", regex ="-.*")
+
+  # some people enter 9-digit ZIPs, which could make them easily identifiable in
+  # the individual output files. rather than truncating to 5 digits -- which may
+  # turn nonsense entered by some respondents into a valid ZIP5 -- we simply
+  # replace these ZIPs with NA.
+  data$zip5 <- ifelse(nchar(data$zip5) > 5, NA_character_,
```

Comment:
The logic for these two functions is borrowed from `delphiFacebook/R/responses.R`, but I'm wondering if this line is supposed to be handling A3. We drop `zip5` anyway, so nulling it out here seems unnecessary. @capnrefsmmat

Reply:
`filter_complete_responses` does the other half of the work: 

https://github.com/cmu-delphi/covidcast-indicators/blob/3590e64ea10aade257030b6f8f0815f9ccc3f30b/facebook/delphiFacebook/R/responses.R#L772-L777

We join with the ZIP metadata based on zip5, then blank out A3 based on the results (not zip5). Don't ask me why we need to use two different columns; undoubtedly the code did that in early 2020 and the logic just got ported over to this version